org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -239,10 +249,15 @@ private void appendGuessingCompletion(StringBuilder buffer, CompletionProposal p
 			}
 
 			char[] argument = parameterNames[i];
-
-			buffer.append("{{");
+			if(supportSnippets){
+				buffer.append("${");
+				buffer.append(Integer.toString(i+1));
+				buffer.append(":");
+			}
 			buffer.append(argument);
```

Comment:
I tried this out with a non vs-code client (the ycmd client I am writing). What this does is that it includes the arguments in the completion string. This is not what the user wants to insert for a method suggestion. 

For the following code:

```java
public class Test {
    public static void main( String[] args ) {
      args./* Completion here */
    }
}
```

I am offered (amongst other things) `wait( long timeout, int nanos ) : void` - one selected, i expect to insert the word `wait`. however, the inserted word is actually `wait( timeout, nanos );` which breaks the user's mental model (it also inserts a `;` which i don't really follow, but is perhaps correct in some sense).

Here's the messages:

```
2017-02-28 20:10:34,998 - DEBUG - TX: Sending message "Content-Length: 221\r\nContent-Type: application/vscode-jsonrpc;charset=utf8\r\n\r\n{\"jsonrpc\": \"2.0\", \"params\": {\"position\": {\"line\": 2, \"character\": 11}, \"textDocument\": {\"uri\": \"file:///Users/ben/Development/java/test/com/puremourning/test/Test.java\"}}, \"id\": \"50\", \"method\": \"textDocument/completion\"}"
...
2017-02-28 20:08:23,152 - DEBUG - RX: Received message: {u'jsonrpc': u'2.0', u'id': u'38', u'result': {u'isIncomplete': False, u'items': [{u'sortText': u'zzzzzzzcba', u'kind': 5, u'data': {u'rid': u'3', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'pid': u'0', u'decl_signature': u'[Ljava.lang.String;', u'name': u'length'}, u'detail': u'String[]', u'label': u'length : int'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'clone', u'pid': u'1', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'[Ljava.lang.String;', u'signature': u'()[Ljava.lang.String;', u'rid': u'3'}, u'detail': u'String', u'label': u'clone() : String[]'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'wait', u'pid': u'2', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'(JI)V', u'rid': u'3'}, u'detail': u'Object', u'label': u'wait(long timeout, int nanos) : void'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'wait', u'pid': u'3', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'(J)V', u'rid': u'3'}, u'detail': u'Object', u'label': u'wait(long timeout) : void'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'wait', u'pid': u'4', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'()V', u'rid': u'3'}, u'detail': u'Object', u'label': u'wait() : void'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'toString', u'pid': u'5', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'()Ljava.lang.String;', u'rid': u'3'}, u'detail': u'Object', u'label': u'toString() : String'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'notifyAll', u'pid': u'6', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'()V', u'rid': u'3'}, u'detail': u'Object', u'label': u'notifyAll() : void'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'notify', u'pid': u'7', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'()V', u'rid': u'3'}, u'detail': u'Object', u'label': u'notify() : void'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'hashCode', u'pid': u'8', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'()I', u'rid': u'3'}, u'detail': u'Object', u'label': u'hashCode() : int'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'getClass', u'pid': u'9', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'()Ljava.lang.Class<*>;', u'rid': u'3'}, u'detail': u'Object', u'label': u'getClass() : Class<?>'}, {u'sortText': u'zzzzzzzadf', u'kind': 3, u'data': {u'name': u'equals', u'pid': u'10', u'uri': u'file:/Users/ben/Development/java/test/com/puremourning/test/Test.java', u'decl_signature': u'Ljava.lang.Object;', u'signature': u'(Ljava.lang.Object;)Z', u'rid': u'3'}, u'detail': u'Object', u'label': u'equals(Object obj) : boolean'}]}}

...

2017-02-28 20:08:23,213 - DEBUG - TX: Sending message "Content-Length: 376\r\nContent-Type: application/vscode-jsonrpc;charset=utf8\r\n\r\n{\"jsonrpc\": \"2.0\", \"params\": {\"sortText\": \"zzzzzzzadf\", \"kind\": 3, \"data\": {\"name\": \"wait\", \"pid\": \"2\", \"uri\": \"file:/Users/ben/Development/java/test/com/puremourning/test/Test.java\", \"decl_signature\": \"Ljava.lang.Object;\", \"signature\": \"(JI)V\", \"rid\": \"3\"}, \"detail\": \"Object\", \"label\": \"wait(long timeout, int nanos) : void\"}, \"id\": \"41\", \"method\": \"completionItem/resolve\"}"

2017-02-28 20:08:23,220 - DEBUG - RX: Received message: {u'params': {u'message': u'28-Feb-2017 20:07:46 >> document/resolveCompletionItem', u'type': 3}, u'jsonrpc': u'2.0', u'method': u'window/logMessage'}

2017-02-28 20:08:23,242 - DEBUG - RX: Received message: {u'jsonrpc': u'2.0', u'id': u'41', u'result': {u'sortText': u'zzzzzzzadf', u'kind': 3, u'documentation': u"Causes the current thread to wait until another thread invokes the `java.lang.Object.notify()` method or the `java.lang.Object.notifyAll()` method for this object, or some other thread interrupts the current thread, or a certain amount of real time has elapsed.\n\nThis method is similar to the wait method of one argument, but it allows finer control over the amount of time to wait for a notification before giving up. The amount of real time, measured in nanoseconds, is given by:\n\n>     1000000*timeout+nanos\n\nIn all other respects, this method does the same thing as the method `wait(long)` of one argument. In particular, wait(0, 0) means the same thing as wait(0).\n\nThe current thread must own this object's monitor. The thread releases ownership of this monitor and waits until either of the following two conditions has occurred:\n\n *  Another thread notifies threads waiting on this object's monitor to wake up either through a call to the notify method or the notifyAll method.\n *  The timeout period, specified by timeout milliseconds plus nanos nanoseconds arguments, has elapsed.\n\nThe thread then waits until it can re-obtain ownership of the monitor and resumes execution.\n\nAs in the one argument version, interrupts and spurious wakeups are possible, and this method should always be used in a loop:\n\n    synchronized (obj) {\n             while (<condition does not hold>)\n                 obj.wait(timeout, nanos);\n             ... // Perform action appropriate to condition\n         }\n\nThis method should only be called by a thread that is the owner of this object's monitor. See the notify method for a description of the ways in which a thread can become the owner of a monitor.\n\n *  **Parameters:**\n    \n     *  **timeout** the maximum time to wait in milliseconds.\n     *  **nanos** additional time, in nanoseconds range 0-999999.\n *  **Throws:**\n    \n     *  IllegalArgumentException if the value of timeout is negative or the value of nanos is not in the range 0-999999.\n     *  IllegalMonitorStateException if the current thread is not the owner of this object's monitor.\n     *  InterruptedException if any thread interrupted the current thread before or while the current thread was waiting for a notification. The *interrupted status* of the current thread is cleared when this exception is thrown.", u'detail': u'Object', u'label': u'wait(long timeout, int nanos) : void', u'textEdit': {u'newText': u'wait(timeout, nanos);', u'range': {u'start': {u'line': 2, u'character': 11}, u'end': {u'line': 2, u'character': 11}}}, u'insertTextFormat': 1}}

```

Is it possible to just get the method name?